samples/tcpsrv.lua

```diff
@@ -0,0 +1,54 @@
+#! /usr/bin/env lua
+
+--
+-- Sample TCP echo server. Listens on 3333 port
+-- Use netcat to send strings to server (ncat 127.0.0.1 3333)
+--
+
+local lgi = require 'lgi'
+local Gio = lgi.Gio
+
+local app = Gio.Application{application_id = 'org.test.tcptest',
+                            flags = Gio.ApplicationFlags.NON_UNIQUE}
```

Comment:
Why is this a `Gio.Application`? I have to admit, I do not really know what this is doing, but I would just remove the app.
(Especially with the `NON_UNIQUE` argument... what does `Gio.Application` do?)

Reply:
Honestly everything should be using `Gio.Application`, manually dealing with the mainloop isn't helpful.

Reply:
Well, what does `GApplication` do that `GLib.MainLoop.new():run()` does not do? I really have no idea.

Reply:
Most examples use `Gio.Application`. Probably because syntax is the same as for `Gtk.Application`.

Reply:
In the context of a demo, nothing. In real world applications, single instance support, daemonization, argument parsing, gresource loading, its just the API applications should use.